### PR TITLE
Add index alias 'project.{index}' into index templates. Closes #12

### LINKED ISSUE #12: Add index aliases "project.{index}" and ".all" into index templates
Add an alias to the index template so that when someone queries for index name `someprojectname.*` it will look in `project.someprojectname.*`.

This should be as simple as adding:

```
"aliases" : {
  "project.{index}" : {}
}
```

into existing index templates. See [indices templates docs](https://www.elastic.co/guide/en/elasticsearch/reference/2.3/indices-templates.html).

### PR DESCRIPTION
**Pull request by lukas-vlcek**
@richm @t0ffel PTAL

### CODE REVIEW

**Comment by richm** [comment]
lgtm

File: viaq-openshift/skeleton.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
-    "aliases": {},
+    "aliases": {
+        "project.{index}" : {}
```
Comment (portante) [inline viaq-openshift/skeleton.json]: Does this work for ES 1.5 and later?

Comment (richm) [reply viaq-openshift/skeleton.json]: If it only works for ES 2.3 and later, that's fine.

Comment (richm) [reply viaq-openshift/skeleton.json]: If this is working, go ahead and merge it, and update https://github.com/openshift/origin-aggregated-logging/pull/222

**Comment by lukas-vlcek** [comment]
@portante Yes, it seems to work for ES `1.5.2` and later.

See my simple reproduction script below:

``` json
$ cat index_template.json 
{
  "template": "*",
  "aliases": {
    "project.{index}" : {},
    ".all": {}
  },
  "mappings": {
  }
}
```

``` shell
$ cat test.sh 
#!/bin/bash
set -o xtrace

# Data and index templates cleanup
curl -X DELETE 'localhost:9200/*'
curl -X DELETE 'localhost:9200/_template/index_template'

# Index first document without index template (we expect no index aliases to be created for it)
curl -X POST 'localhost:9200/index1/type1' -d '{"foo":"bar"}'

# Add index template to create index aliases
curl -X PUT 'localhost:9200/_template/index_template' -d@index_template.json
curl -X GET 'localhost:9200/_template/index_template?pretty'

# Index second document
curl -X POST 'localhost:9200/index2/type1' -d '{"foo":"bar"}'

# Make sure data are fully indexed
curl -X POST 'localhost:9200/_refresh'

# See what has happened
curl -X GET 'localhost:9200/_cat/indices?v'
curl -X GET 'localhost:9200/_cat/aliases?v'
```

Tested with `1.5.2`, `1.7.2`, `2.3.5`, `2.4.0` and `5.0.0-alpha5`.

It works consistently, see below the end of the test script output:

```
curl -X GET 'localhost:9200/_cat/indices?v'
health status index  uuid                   pri rep docs.count docs.deleted store.size pri.store.size
yellow open   index1 3nyRDNqWRt6qRnZWBwKWGw   5   1          1            0       260b           260b
yellow open   index2 CQuAOYUwSL-KYyTuXyNZeg   5   1          1            0       260b           260b

curl -X GET 'localhost:9200/_cat/aliases?v'
alias          index  filter routing.index routing.search
.all           index2 -      -             -
project.index2 index2 -      -             -
```

We can see that `index1` has no aliases attached to it, while `index2` has expected index aliases. In this case aliases `project.index2` and `.all` (I am adding the later one as per [other](https://github.com/openshift/origin-aggregated-logging/pull/222#issuecomment-249879243) @ewolinetz request).

**Comment by lukas-vlcek** [comment]
Do not merge for now.

**Comment by t0ffel** [comment]
this is inverse of what is needed.
we need to be able to derive alias from index name as follows:
index name: project.<project_name>.<project_uuid>-YYYY.MM.DD
alias: <project_name>.<project_uuid>-YYYY.MM.DD

**Comment by lukas-vlcek** [comment]
@t0ffel yea, it seems this is inverse of what we really need (that is why I added the DO NOT MERGE label). Unfortunately, index templates are not very flexible about naming index aliases, it can not do much beyond adding hardcoded strings to index names.

**Comment by richm** [comment]
Can we close this now?

**Comment by lukas-vlcek** [comment]
@richm Before we close this we need to figure out which commit the existing proposed common data model index templates for this PR (https://github.com/openshift/origin-aggregated-logging/pull/222) come from.

That mentioned PR still includes the **`.all` alias** for two index templates, one for **operations** and one for **projects**.

So we can open a new PR for **ViaQ/elasticsearch-templates** that will only add `.all` alias, however, at this moment I can not open it against master branch, because it has been already adopted to "one type per index" model (is that correct?). So may be we can created a branch just for the sake of OSE release that will include common data model and then open new PR against this branch to add `.all` alias. This branch should probably be based on this commit: https://github.com/ViaQ/elasticsearch-templates/commit/774ac92541642b82e40c0dac1015b993debe6bf1?

WDYT?

**Comment by lukas-vlcek** [comment]
@richm ok, so we need to talk :-) I can see there were some changes in common data model in the mean-time.

**Comment by lukas-vlcek** [comment]
I am closing this ticket. Common Data Model has been updated in the mean-time. My only concern is if we need to use the `.all` alias of not. @richm let me know if we need to add this alias.

**Comment by richm** [comment]
The `.all` alias is used by the openshift-elasticsearch-plugin.

**Comment by richm** [comment]
but it looks like the plugin will create the `.all` alias if needed: https://github.com/fabric8io/openshift-elasticsearch-plugin/blob/master/src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java#L228

**Comment by ewolinetz** [comment]
@richm with the creation of the `.all` alias with the index templates I plan on phasing that code out since its pretty hackish and there are probably scenarios where you could miss out on data from there

**Comment by richm** [comment]
ok - if we don't need the `.all` index, then we don't need it

**Pull request closed without merging**